### PR TITLE
v4.2.0 - Audio Library Changes, Many Plugin Updates, Etc.

### PR DESCRIPTION
**Pull request by DuckBoss**
### tl;dr uninstall vlc, install ffmpeg

<br>
<br>

### Experimental FFMPEG Library Testing
- Experimenting with using FFMPEG directly instead of VLC.
- This change should theoretically not be noticeable to any users.

### Bot Commands Plugin
- Added new command: !cmdsearch
    -  Search for a command from the full list of commands available using fuzzy-searching to find the closest results.
    `!cmdsearch 'command_name' : Search for a command from the full list of commands available.`

### Sound Board Plugin
- !sbdownload command now supports an optional parameter providing a timestamp range to trim the downloaded clip.
`!sbdownload 'youtube_link' 'file_name' 'H:M:S-H:M:S'`
Example:
`!sbdownload https://youtu.be/bqDrftAxYpk test_file 0:1:10-0:2:15`
- Added new alias: sbd - !sbdownload
`!sbd 'youtube_link' 'file_name'`
- sound_board files played with !sb/!sbnow/!sbquiet/!sbquietnow can now loop tracks when loop mode is enabled.
    - sound_board files played with !sbrandom/!sbrandomnow don't loop.

### Images Plugin
- Added new command: !imgsearch
    Search for a command from the full list of images available using fuzzy-searching to find the closest results.
    - `!imgsearch 'image_name': Searches through locally stored images and returns the closest results.`
- Fixed !post command.
- The !post command now displays an error message when a common http error occurs.

### Server Tools Plugin
The server_tools plugin is an administrative plugin for high-permission users to set callback methods to mumble server events.
For now, the plugin contains callback events to play an audio clip when users join the server. This is an optional feature, configurable in the plugins' metadata.ini file.
**Note: Any audio clips from the sound_board permanent directory can be used.**
**Note: The user connection audio feature is disabled by default and can be turned on in the metadata.ini file**

- **Supports generic connection sound clips for all users, and per-user connection sound clips**
    - Added new command: !setdefaultconnectionsound
    `!setdefaultconnectionsound 'audio_clip_name': Sets the generic audio clip that is played when a user joins the server.`
    - Added new command: !clearuserconnectionsound
    `!clearuserconnectionsound 'username': Clears the audio clip associated with the given username.`
    - Added new command: !getuserconnectionsound 
    `!getuserconnectionsound 'username': Displays the audio clip set to play whenever the given username connects to the server.`
    - Added new command: !setuserconnectionsound 
    `!setuserconnectionsound 'username' 'audio_clip_name': Sets the audio clip that will play whenever the given user connects to the server.`

### Other Updates
- Updated versions to v4.2.0
- Updated youtube_dl dependency version to 2020.9.14

### CODE REVIEW

File: JJMumbleBot/plugins/extensions/sound_board/help.html [hunk 1]
```diff
@@ -8,4 +8,5 @@
 <b>!sbsearch 'file_name'</b>: Finds locally saved audio clips by best match and displays the results in a list.<br>
 <b>!sblist/!sbl</b>: Displays the full exhaustive list of all the available sound board clips in a private message.<br>
 <b>!sbdownload 'youtube_url' 'file_name'</b>: Downloads a sound clip from a youtube link and adds it to the sound board.<br>
+<b>!sbdownload 'youtube_url' 'file_name' 'XXh:XXm:XXs-XXh:XXm:XXs'</b>: Downloads a sound clip from a youtube link and trims it to the given time frame and adds it to the sound board.<br>
```
Comment (kaminascripts) [inline JJMumbleBot/plugins/extensions/sound_board/help.html:11]:
```suggestion
<b>!sbdownload 'youtube_url' 'file_name' 'H:M:S-H:M:S'</b>: Downloads a sound clip from a youtube link and trims it to the given time frame and adds it to the sound board.<br>
```

**Comment by kaminascripts** [comment]
Just a tip about the documentation update process, what do you think about moving the wiki to a doc folder inside the repository root with a TOC ? I think this would ease to update documentation because you currently have to create a pull request with the solution and then go to the wiki and do the update without patch relation, if you move the documentation you can change it in the same PR or in another related PR and also you can share them and ask for reviewers if you want.

**Comment by DuckBoss** [comment]
> Just a tip about the documentation update process, what do you think about moving the wiki to a doc folder inside the repository root with a TOC ? I think this would ease to update documentation because you currently have to create a pull request with the solution and then go to the wiki and do the update without patch relation, if you move the documentation you can change it in the same PR or in another related PR and also you can share them and ask for reviewers if you want.

Yes that's a good point.
I could do this, but it would take quite a bit of time to move everything over into a web page format
at https://duckboss.github.io/JJMumbleBot/
